torch/_dynamo/compiled_autograd.py

```diff
@@ -21,15 +22,33 @@
 )
 from torch.fx.experimental.symbolic_shapes import DimDynamic, ShapeEnv
 from torch.fx.proxy import Proxy
+from torch._functorch.aot_autograd import AOT_COUNTER
 
 compiled_autograd_log = getArtifactLogger(__name__, "compiled_autograd")
 
-
 def maybe_clone(x):
     if x is not None:
         return clone_preserve_strides(x)
     return x
 
+class CompiledAutogradCustomFunctionKey:
+    # Used by user-defined autograd function
+    # Additionally guard against their saved tensors shapes
+    COMPILED_AUTOGRAD_ID = None
+    CURRENT_SAVED_TENSORS_SHAPES = ()
```

Comment:
Why is this needed?

Reply:
removed, we already collect saved tensors via TensorArgs::id